src/polyfillstub/intersection-observer-stub.js

```diff
@@ -39,10 +39,25 @@ export function shouldLoadPolyfill(win) {
     !win.IntersectionObserver ||
     !win.IntersectionObserverEntry ||
     !!win.IntersectionObserver[STUB] ||
-    !supportsDocumentRoot(win)
+    !supportsDocumentRoot(win) ||
+    isWebkit(win)
   );
 }
 
+/**
+ * All current (as of Safari 14.x) {root:document} IntersectionObservers
+ * will report incorrect rootBounds, intersectionRect, and intersectionRatios
+ * and therefore we force the polyfill in this case.
+ * See: https://bugs.webkit.org/show_bug.cgi?id=219495.
+ *
+ * @param {!Window} win
+ * @return {boolean}
+ */
+function isWebkit(win) {
+  const ua = win.navigator && win.navigator.userAgent;
+  return /WebKit/.test(ua);
```

Comment:
This will catch Android's too, no?

Reply:
> Note: I've never written UA engine detection code before, so I'm still researching now to ensure WebKit is safe as a regex test.

Yeah, that strategy was total garbage.  Hopefully I can rely on the actual Platform service? The Safari detection in particular is difficult since it relies on ruling out all of the other browsers 😂

Reply:
Unfortunately, I don't think we can rely on services inside a polyfill - those go ahead of everything else. Might just need to fork some of those checks, though. Or maybe there's a better check that someone has online.

Reply:
I think I found a better one for specifically testing iOS/Webkit vs. everything else

Reply:
Awesome!